File: core/src/ru/myitschool/dcrawler/entities/skills/action/SimpleCombinedAction.java
package ru.myitschool.dcrawler.entities.skills.action;

import ru.myitschool.dcrawler.math.MathAction;
import ru.myitschool.dcrawler.math.NumberAction;
import ru.myitschool.dcrawler.entities.skills.FloatingDamageMark;
import ru.myitschool.dcrawler.entities.skills.Skill;
import ru.myitschool.dcrawler.entities.skills.Target;

public abstract class SimpleCombinedAction extends Action {

    private MathAction action;
    private MathAction critAction;

    public SimpleCombinedAction(Skill skill, MathAction action) {
        super(skill);
        this.action = action;
        this.critAction = new NumberAction(action.max());
    }

    @Override
    protected MathAction successDamage() {
        return null;
    }

    @Override
    protected MathAction critSuccessDamage(int standardDamage) {
        return null;
    }

    @Override
    protected MathAction failDamage() {
        return null;
    }

    @Override
    protected MathAction critFailDamage(int standardDamage) {
        return null;
    }


    @Override
    protected void critSuccessEffect(Target target, int damage, FloatingDamageMark mark) {

    }

    @Override
    protected void failEffect(Target target, int damage, FloatingDamageMark mark) {

    }

    @Override
    protected void critFailEffect(Target target, int damage, FloatingDamageMark mark) {

    }

    @Override
    protected void beforeEffect(Target target, int damage, FloatingDamageMark mark) {

    }

    @Override
    protected void afterEffect(Target target, int damage, FloatingDamageMark mark) {

    }
}

File: core/src/com/wolg_vlad/dcrawler/entities/skills/action/SimpleCombinedAction.kt
package com.wolg_vlad.dcrawler.entities.skills.action

import com.wolg_vlad.dcrawler.entities.skills.FloatingDamageMark
import com.wolg_vlad.dcrawler.entities.skills.Skill
import com.wolg_vlad.dcrawler.entities.skills.Target
import com.wolg_vlad.dcrawler.math.MathAction
import com.wolg_vlad.dcrawler.math.NumberAction

abstract class SimpleCombinedAction(skill: Skill, private val action: MathAction) : Action(skill) {
    private val critAction: MathAction
    override fun successDamage(): MathAction? {
        return null
    }

    override fun critSuccessDamage(standardDamage: Int): MathAction? {
        return null
    }

    override fun failDamage(): MathAction? {
        return null
    }

    override fun critFailDamage(standardDamage: Int): MathAction? {
        return null
    }

    override fun critSuccessEffect(target: Target?, damage: Int, mark: FloatingDamageMark?) {}
    override fun failEffect(target: Target?, damage: Int, mark: FloatingDamageMark?) {}
    override fun critFailEffect(target: Target?, damage: Int, mark: FloatingDamageMark?) {}
    override fun beforeEffect(target: Target?, damage: Int, mark: FloatingDamageMark?) {}
    override fun afterEffect(target: Target, damage: Int, mark: FloatingDamageMark) {}

    init {
        critAction = NumberAction(action.max())
    }
}